src/main/java/com/sap/oss/phosphor/fosstars/tool/github/RatingCalculator.java

```diff
@@ -33,5 +35,17 @@
    * @return The same calculator.
    * @throws IOException If something went wrong.
    */
-  RatingCalculator calculateFor(GitHubProject project) throws IOException;
+  default RatingCalculator calculateFor(GitHubProject project) throws IOException {
```

Comment:
Why not make it a default method with the basic implementation?

Reply:
`default` is good when you don't know which classes implements the interface, for example, if there are external projects that use the interface. If the interface gets updated, the external projects fail to compile. To avoid it, `default` may be used. However, `default` makes an interface look more like a regular class since it has method implementations. But an interface should just define clean API. In our case, we have a couple of classes that implements this interface. There is no risk that something stops to compile. To keep the interface clean, we can just implement this method in those classes. Furthermore, I think we pay more attentions to the specific context when we implement a method for a specific class.

Reply:
Valid point, I like to use default methods to provide an implementation for the default case and only someone who implements the interface and need something different can just override it.
Most times this avoids duplicate code if only the default case is required.

But I align it with the projects coding style 🙂

Reply:
Valid point, I like to use default methods to provide an implementation for the default case and only someone who implements the interface and need something different can just override it.
Most times this avoids duplicate code if only the default case is required.

But I align it with the projects coding style 🙂

Reply:
Thanks. If you see duplicate code, we can introduce an abstract common class.